src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -0,0 +1,158 @@
+        fatal_error_reporter: a queue to report fatal errors back to the main process
+        testing_queue: queue used to send commands to the simulator. Should only be used in unit tests
+        read_timeout_seconds: number of seconds to wait until read is of desired size before returning how ever many bytes have been read. Timeout should be set to 0 unless a non-zero value is necessary for unit testing
+    """
+
+    def __init__(
+        self,
+        input_queue: Queue[
+            bytes
+        ],  # pylint: disable=unsubscriptable-object # https://github.com/PyCQA/pylint/issues/1498
+        output_queue: Queue[bytes],  # pylint: disable=unsubscriptable-object
+        fatal_error_reporter: Queue[  # pylint: disable=unsubscriptable-object
+            Dict[str, Any]
+        ],
+        testing_queue: Queue[Dict[str, Any]],  # pylint: disable=unsubscriptable-object
+        logging_level: int = logging.INFO,
+        read_timeout_seconds: Union[int, float] = 0,
+    ) -> None:
+        super().__init__(fatal_error_reporter, logging_level=logging_level)
+        self._output_queue = output_queue
+        self._input_queue = input_queue
+        self._testing_queue = testing_queue
+        self._init_time_ns: Optional[int] = None
+        self._time_of_last_status_beacon_secs: Optional[float] = None
+        self._leftover_read_bytes: Optional[bytes] = None
+        self._read_timeout_seconds = read_timeout_seconds
+
+    def _setup_before_loop(self) -> None:
+        # Tanner (2/2/21): Comparing perf_counter_ns values in a subprocess to those in the parent process have unexpected behavior in windows, so storing the initialization time after the process has been created in order to avoid issues
+        self._init_time_ns = perf_counter_ns()
+
+    def get_cms_since_init(self) -> int:
+        if self._init_time_ns is None:
+            return 0
+        ns_since_init = perf_counter_ns() - self._init_time_ns
+        return ns_since_init // NANOSECONDS_PER_CENTIMILLISECOND
+
+    def _get_cms_timestamp_bytes(self) -> bytes:
+        return self.get_cms_since_init().to_bytes(8, byteorder="little")
+
+    def _send_status_beacon(self, truncate: bool = False) -> None:
+        self._time_of_last_status_beacon_secs = perf_counter()
+        num_bytes_in_content = b"\x0e\x00"
+        module_id = b"\x00"
+        packet_type = b"\x00"
+
+        status_beacon = SERIAL_COMM_MAGIC_WORD_BYTES
+        status_beacon += num_bytes_in_content
+        status_beacon += self._get_cms_timestamp_bytes()
+        status_beacon += module_id
+        status_beacon += packet_type
+        status_beacon += _get_checksum_bytes(status_beacon)
+        if truncate:
+            trunc_index = random.randint(  # nosec B311 # Tanner (2/4/21): Bandit blacklisted this psuedo-random generator for security/cryptographic reasons which do not apply to the desktop app.
+                0, 10
```

Comment:
why did you choose `10`? right now it's a bit of a magic number just looking at the code. If we are looking at the code in the future, it'd be good to know why.  Is this one less than the total length of the beacon?